### ISSUE TITLE
discriminator_model trainable

### ISSUE THREAD

**Issue by mnill** [rickyhan/SimGAN-Captcha#1]
U set discriminator_model.trainable = False, but never toggle it back, is it ok?

As i known, code must look like 
    discriminator_model.trainable = True
    disc_loss_real = discriminator_model.train_on_batch(real_image_batch, y_real)
    disc_loss_refined = discriminator_model.train_on_batch(refined_image_batch, y_fake)
    discriminator_model.trainable = False
    combined_loss = combined_model.train_on_batch(synthetic_image_batch, [synthetic_image_batch, y_real])

**Comment by rickyhan** [comment]
This is the intended behavior. Please take a look at this issue:

https://github.com/fchollet/keras/issues/4674#issuecomment-286074399

**Comment by mnill** [comment]
Thanks

**Comment by mnill** [comment]
https://github.com/keras-team/keras/issues/8585

**Comment by davidlight2018** [comment]
hello @mnill @rickyhan , 
I have the same question. Is it ok not to do as you said? Like:
`discriminator_model.trainable = True
disc_loss_real = discriminator_model.train_on_batch(real_image_batch, y_real)
disc_loss_refined = discriminator_model.train_on_batch(refined_image_batch, y_fake)
discriminator_model.trainable = False
combined_loss = combined_model.train_on_batch(synthetic_image_batch, [synthetic_image_batch, y_real])`

As I noticed after 2000 epochs, my "Discriminator model loss refined" still remain the same as first few rounds training at 0.69314, it did not decrease at all. So is there any suggestions? Thanks a lot.

**Comment by davidlight2018** [comment]
And after gan training, I got all predict results is 0, any ideas?
```
synthetic_image_batch = get_image_batch(synth_generator())
arr = refiner_model.predict_on_batch(synthetic_image_batch)
discriminator_model.predict(arr)
```
